src/hooks/use-text-filter.js

```diff
@@ -10,11 +12,78 @@ const DEBOUNCE_TIME = 200;
  * postprocessing of data from Realm
  * @param {*} query the text filter query to use for searching
  */
-function useTextFilter(query) {
+function useTextFilter(query, activeContentTab) {
     const [filterEvent, setFilterEvent] = useState(null);
     const [results, setResults] = useState(null);
+    /**
+     * Gets all results
+     * @param {Array} filterResults
+     * @returns {Array}
+     */
+    const getAll = filterResults => {
+        return filterResults.map(r => {
+            switch (r.mediaType) {
+                case 'youtube':
+                case 'twitch':
+                    return {
+                        ...new SearchVideoResult(r),
+                    };
+                case 'podcast':
+                    return {
+                        ...new SearchPodcastResult(r),
+                    };
+                default:
+                    return {
+                        ...new SearchArticleResult(r),
+                    };
+            }
+        });
+    };
+
+    /**
+     * Extracts only articles from filterResults
+     * @param {Array} filterResults
+     * @returns {Array}
+     */
+    const getArticles = filterResults => {
+        return filterResults.reduce((results, r) => {
+            if (r.mediaType === 'article') {
+                results.push({ ...new SearchArticleResult(r) });
+            }
+            return results;
+        }, []);
+    };
+
+    /**
+     * Extracts only twitch or youtube videos from filterResults
+     * @param {Array} filterResults
+     * @returns {Array}
+     */
+    const getVideos = filterResults => {
+        return filterResults.reduce((results, r) => {
+            if (r.mediaType === 'twitch' || r.mediaType === 'youtube') {
+                results.push({ ...new SearchVideoResult(r) });
+            }
+            return results;
+        }, []);
+    };
+
+    /**
+     * Extracts only articles from filterResults
+     * @param {Array} filterResults
+     * @returns {Array}
+     */
+    const getPodcasts = filterResults => {
+        return filterResults.reduce((results, r) => {
+            if (r.mediaType === 'podcast') {
+                results.push({ ...new SearchPodcastResult(r) });
+            }
+            return results;
+        }, []);
+    };
```

Comment:
There is a decent amount of repeated logic, can we generalize a bit and pass media types to check and the class as arguments into a general function? (Or use that to generate more specific functions).

What fo you think?

Reply:
It might be a little tricky to do because the if in line 64 checks for two things `twitch` or `youtube` instead of one thing like the rest